### PR TITLE
exclude ##.adsbygoogle

### PR DESCRIPTION
**Pull request by Yuki2718**
For https://github.com/easylist/easylist/commit/c09ae084dc666529d0ae0e639836f423bd1231ff

BTW isn't `^` needed for `/###banner_ad$/` as well? Specific `###banner_ad` may be used by good reason.

### CODE REVIEW

**Comment by AdamWr** [comment]
I guess it's `/###banner_ad$/`, because there is a rule like this `~bg-ikonomika.com###banner_ad` in the `Bulgarian list`.
Maybe it should be changed to something like - `/^(~.*)?###banner_ad$/`?

By the way, now I have noticed that also `ROList` has rule like this `~4tuning.ro##.adsbygoogle` and `xinggsf` has `~google.com##.adsbygoogle`

So maybe it also should be - `/^(~.*)?##\.adsbygoogle$/`?

**Review by AdamWr — APPROVED** [review]

**Review by zloyden — APPROVED** [review]

**Review by piquark6046 — APPROVED** [review]

**Comment by Yuki2718** [comment]
Now I think of, there can theoretically be this type of specific filter: `~a.foo.com,foo.com##.adsbygoogle`
So I suggest to change to `/^(~[-.a-z]+,?)*##\.adsbygoogle$/`

**Comment by AdamWr** [comment]
That's a good point, but if I'm not wrong, it will not cover all cases.
For example, it will not match domains with digits or with "punycode".
Maybe something like this:
```
^(~([-.\w\d*]|[^\x00-\x7F])+,?)*##\.adsbygoogle$
```
will be okay (https://regex101.com/r/LLqMr0/1), but I'm still not sure if all cases will be catched.

File: filters/exclusions.txt [hunk 1]
```diff
@@ -36,9 +36,10 @@ $popup,~third-party
 !
 !### Anti-adblock baits ###
 ! all filters
-/###banner_ad$/
-/^##\.adsbox$/
-/^##\.afs_ads$/
+/^(~.*)?##\.adsbygoogle$/
+/^(~.*)?###banner_ad$/
+/^(~.*)?##\.adsbox$/
+/^(~.*)?##\.afs_ads$/
```
Comment (Alex-302) [inline filters/exclusions.txt:42]: why changed?

Comment (Yuki2718) [reply filters/exclusions.txt:42]: Can there be harm? Some filter may add `~foo.com##.afs_ads` in future.

Comment (Alex-302) [reply filters/exclusions.txt:42]: Ok.

**Pull request merged**